### PR TITLE
[WIP] Make Google Analytics tracker(s) configurable

### PR DESCRIPTION
**Pull request by adamzerella**
### What this PR does

WIP for #1713 allowing a data admin to access the Google Analytics identifier set in Helm, assuming `admin-api` is a suitable module. 

`admin-api` helm chart is configured from static value --> `magda-admin-api` accepts helm input --> `magda-admin-api` offers API endpoint `/gapi`.

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable
-   [x] I've updated CHANGES.md with what I changed.
-   [ ] I've linked this PR to an issue in ZenHub (core dev team only)
-   [x] I've assigned this PR to someone (if you don't know who to assign it to, pick @AlexGilleran)

### CODE REVIEW

**Comment by AlexGilleran** [comment]
Great to see you again @adamzerella 😄.

> assuming admin-api is a suitable module.

It _kinda_ makes sense, but I think it'd be easier to just have it supplied to the `magda-web-server` module instead and then passed to the client with the rest of the config, seeing how it's only the client that's going to use it.

**Comment by adamzerella** [comment]
@AlexGilleran what's crackin' :wave: 

Yeah I think the wording of the issue threw me off a little, I've moved the code into the `web-server` module and added a test for it. Haven't tested locally yet, hopefully will soon

**Comment by AlexGilleran** [comment]
OK that's a bit better, but rather than provide an endpoint specifically for this I'd just put it in the `webServerConfig` that the client javascript (`magda-web-client`) is already grabbing on startup.

Also keep in mind that it needs to be connected up in `magda-web-client` - right now that's at https://github.com/magda-io/magda/blob/master/magda-web-client/public/assets/third-party.js#L34 but it should probably done better - we should just be downloading the Google Analytics code once and configuring it with one or more Google Analytics ids as passed through the config from the web server pod rather than relying on pasting code snippets.

**Comment by adamzerella** [comment]
@AlexGilleran I've swapped around this PR a little, the `webServerConfig` config now accepts a list of tracker ids that are passed into `react-ga`; This module seems to accept an array of tracker ids: https://github.com/react-ga/react-ga/blob/92205207684e5df19c172f93f7a9e1d0b3edb179/types/index.d.ts#L85

I've refactored and removed the current implementation of sending `ga` events in favour of using `react-ga`. Still needs to be tested on deployment as I'm not sure a `.yml` list value will translate to js well.

Let me know if you prefer the `magda-web-client` to handle ga / ga events another way

File: magda-web-client/src/AppContainer.js [hunk 1]
```diff
@@ -23,6 +24,28 @@ import "./AppContainer.css";
 class AppContainer extends React.Component {
     componentDidMount() {
         this.props.requestWhoAmI();
+
+        const _trackers = [];
+
+        // Create GA trackers from the list provided to us
+        config.gapiId
```
Comment (adamzerella) [inline magda-web-client/src/AppContainer.js]: Should clean this up a little and discuss if better places to put this

File: magda-web-client/public/assets/third-party.js [hunk 2]
```diff
@@ -31,8 +31,6 @@
     "ga"
 );
```
Comment (AlexGilleran) [inline magda-web-client/public/assets/third-party.js]: You can get rid of the google analytics snippet above too :)

File: magda-web-client/src/AppContainer.js (hunk 1 above)
Comment (AlexGilleran) [reply magda-web-client/src/AppContainer.js]: Yeah I think just refactor it into some kind of `initGoogleAnalytics.js` file that exports a function that does this.

I'm a bit confused by the use of the `?` and the `[config.gapiId]` thing here though - isn't `config.gapiId` already an array?

File: magda-web-server/src/index.ts [hunk 3]
```diff
@@ -168,7 +173,8 @@ const webServerConfig = {
     ),
     fallbackUrl: argv.fallbackUrl,
     datasetSearchSuggestionScoreThreshold:
-        argv.datasetSearchSuggestionScoreThreshold
+        argv.datasetSearchSuggestionScoreThreshold,
+    gapiId: argv.gapiId
```
Comment (AlexGilleran) [inline magda-web-server/src/index.ts]: It'd probably be better to call it `gapiIds` because it's an array :)

File: deploy/helm/magda/values.yaml [hunk 4]
```diff
@@ -13,6 +13,8 @@ global:
   logLevel: INFO
   noDbAuth: false
   namespace: default
+  gapiId: 
+    - UA-92539508-3
```
Comment (AlexGilleran) [inline deploy/helm/magda/values.yaml]: This should probably be `[]` by default then we can add this and the other id in https://github.com/magda-io/magda-config-data.gov.au :)

**Comment by adamzerella** [comment]
@AlexGilleran Updated the PR from your notes, naming convention for the analytics module is in the air.

Maybe making `ga.js` an actual module and tidying it up to something similar to: https://github.com/react-ga/react-ga/blob/master/demo/app/index.jsx is the next step. It seems that we'd need to initialise the component before every call. 

> GA must be initialized using this function before any of the other tracking functions will record any data. The values are checked and sent through to the ga('create', ... call.

From: https://github.com/react-ga/react-ga#reactgainitializegatrackingid-options

Side note, I'll open another PR to add `gapIds` to https://github.com/magda-io/magda-config-data.gov.au/blob/master/config.yaml

**Comment by AlexGilleran** [comment]
Made some updates to this but I can't put the comments into here or raise a separate PR, closing this one

**Comment by AlexGilleran** [comment]
OK new PR is up at https://github.com/magda-io/magda/pull/1783/files.

@adamzerella do you want to take a look and see if you're ok with everything I did? Essentially boiled down to:
- Making sure the trackers always got passed to ReactGA
- Smoothing out the helm stuff

Thanks again! 😁

**Comment by AlexGilleran** [comment]
Definitely! 💯

**Pull request closed without merging**